# 1. Business Understanding

# 2. Data Understandind


    Business Understanding
    Data Understanding
    Data Preparation
    Data Analysis
    


In [ ]:
# Imort libraries

In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import scipy.stats as stats
import statsmodels.api as sm

In [8]:
import os
print(os.getcwd())

c:\Users\lenovo\Documents\Moringa ips\Phase 1 ip


In [26]:
aviation_data = pd.read_csv('data/AviationData.csv', encoding='latin1')
state_codes = pd.read_csv('data/USState_Codes.csv', encoding='latin1')

C:\Users\lenovo\AppData\Local\Temp\ipykernel_8416\3935774916.py:1: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  aviation_data = pd.read_csv('data/AviationData.csv', encoding='latin1')


In [27]:
# original data before modifying
aviation_data_raw = aviation_data.copy()
state_codes_raw = state_codes.copy()

In [45]:


import os
print(os.getcwd())
!dir

c:\Users\lenovo\Documents\Moringa ips\Phase 1 ip
 Volume in drive C has no label.
 Volume Serial Number is C20C-5E1E

 Directory of c:\Users\lenovo\Documents\Moringa ips\Phase 1 ip

04/25/2025  12:58 PM    <DIR>          .
04/25/2025  12:10 PM    <DIR>          ..
04/28/2025  03:57 PM             2,319 Aviation_Safety_Assessment..ipynb
04/28/2025  10:30 PM    <DIR>          Data
04/25/2025  12:42 PM    <DIR>          Images
04/25/2025  12:12 PM    <DIR>          Presentation
04/25/2025  12:42 PM                 0 README.md
               2 File(s)          2,319 bytes
               5 Dir(s)  43,262,590,976 bytes free


In [50]:
df1 = pd.read_csv('data/Aviation_Data.csv', encoding='latin1')
df2 = pd.read_csv('data/USState_Codes.csv', encoding='latin1')


C:\Users\lenovo\AppData\Local\Temp\ipykernel_8416\2939004643.py:1: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df1 = pd.read_csv('data/Aviation_Data.csv', encoding='latin1')


In [ ]:
# data exploration

In [ ]:


print(df1.head())
print(df2.head())


         Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1         

In [ ]:
#Merging the two data sets

In [14]:
('Aviation_Data.csv_and_USStates_Codes.csv')

'Aviation_Data.csv_and_USStates_Codes.csv'

In [58]:
df1.info()

df1.shape


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

(88889, 31)

In [61]:
df2.info()
df2.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   US_State      62 non-null     object
 1   Abbreviation  62 non-null     object
dtypes: object(2)
memory usage: 1.1+ KB


(62, 2)

In [ ]:
# dataframe numeric columns

In [56]:
df1.describe()


Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [57]:
df2.describe()

US_State Abbreviation
count        62           62
unique       62           62
top     Alabama           AL
freq          1            1

# 3.data cleaning

Identify Missing Data

Drop Irrelevant Columns

Standardise the data for Analysis

In [66]:
df2 = df2.drop_duplicates()

In [62]:
# number of null values in each column
df1.isnull().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38757
Airport.Name              36185
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1382
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7096
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6384
Publication.Date          13771
dtype: i

In [63]:
df2.isnull().sum()

US_State        0
Abbreviation    0
dtype: int64

In [72]:
# percentage of columns with missing values
missing_percentage = df1.isnull().mean() * 100
missing_percentage = missing_percentage[missing_percentage > 0]
print(missing_percentage)

Location                   0.058500
Country                    0.254250
Latitude                  61.320298
Longitude                 61.330423
Airport.Code              43.601570
Airport.Name              40.708074
Injury.Severity            1.124999
Aircraft.damage            3.593246
Aircraft.Category         63.677170
Registration.Number        1.554748
Make                       0.070875
Model                      0.103500
Amateur.Built              0.114750
Number.of.Engines          6.844491
Engine.Type                7.982990
FAR.Description           63.974170
Schedule                  85.845268
Purpose.of.flight          6.965991
Air.carrier               81.271023
Total.Fatal.Injuries      12.826109
Total.Serious.Injuries    14.073732
Total.Minor.Injuries      13.424608
Total.Uninjured            6.650992
Weather.Condition          5.053494
Broad.phase.of.flight     30.560587
Report.Status              7.181991
Publication.Date          15.492356
dtype: float64


In [ ]:
# # 4.Cleaning the data

In [19]:
threshold = 30

In [26]:
df_cleaned = df.drop(columns=columns_to_drop)
aviation_data_final = df_cleaned

NameError: name 'df' is not defined

In [98]:
df1.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [ ]:
# analyzing safest aircraft model

Idenntify trends, Which aircraft models/types have the fewest accidents?
Compare accident rates over time.
Analyze the severity of incidents by aircraft type.

In [ ]:
# Analyzing safest aircraft model

Fatal accidents (highest risk)
No injuries , good
Minor injuries
Serious injuries




aviation_data_final = df_cleaned
aviation_data_final['Injury.Severity'].value_counts()

In [99]:
aviation_data_final = df_cleaned
aviation_data_final['Injury.Severity'].value_counts()

Injury.Severity
Non-Fatal     67357
Fatal(1)       6167
Fatal          5262
Fatal(2)       3711
Incident       2219
              ...  
Fatal(270)        1
Fatal(60)         1
Fatal(43)         1
Fatal(143)        1
Fatal(230)        1
Name: count, Length: 109, dtype: int64

In [100]:
aviation_data_final['Fatal_Accident'] = aviation_data_final['Injury.Severity'].apply(lambda x: 1 if x == 'Fatal' else 0)
aviation_data_final['Serious_Accident'] = aviation_data_final['Injury.Severity'].apply(lambda x: 1 if x == 'Serious' else 0)

In [106]:
# by make and model
grouped = aviation_data_final.groupby(['Make', 'Model']).agg(Fatal_Accident=('Fatal_Accident', 'sum'), Serious_Accident=('Serious_Accident', 'sum'), Total_Accidents=('Injury.Severity', 'count')).reset_index()
grouped.sort_values(by='Total_Accidents', ascending=False).head(10)


Make      Model  Fatal_Accident  Serious_Accident  Total_Accidents
5745   Cessna        152               2                 0             2168
5767   Cessna        172              26                 0             1253
5811   Cessna       172N               1                 0              996
15079   Piper  PA-28-140               1                 0              812
5720   Cessna        150              10                 0              715
5809   Cessna       172M               1                 0              667
5814   Cessna       172P               0                 0              597
14973   Piper      PA-18               1                 0              539
5744   Cessna       150M               0                 0              539
15088   Piper  PA-28-161               0                 0              502

In [107]:
# safest aircraft
filtered = grouped[grouped['Total_Accidents'] >= 10]
safest_aircraft = filtered.sort_values(by=['Fatal_Accident', 'Serious_Accident', 'Total_Accidents'])
safest_aircraft.head(10)

Make    Model  Fatal_Accident  Serious_Accident  \
258   AIR TRACTOR INC   AT 802               0                 0   
712    Aero Commander     200D               0                 0   
733    Aero Commander      680               0                 0   
741    Aero Commander    680FL               0                 0   
946      Aerospatiale  AS-350B               0                 0   
1007     Aerospatiale  SA-315B               0                 0   
1066         Aerostar     RX-8               0                 0   
1211      Air Tractor  AT-400A               0                 0   
1216      Air Tractor  AT-402A               0                 0   
2145             BELL     47D1               0                 0   

      Total_Accidents  
258                10  
712                10  
733                10  
741                10  
946                10  
1007               10  
1066               10  
1211               10  
1216               10  
2145               10

# 5. Visualize Key Insights

In [108]:
import matplotlib.pyplot as plt

# Create simple, focused  bar graphs 

In [ ]:
# top 10 safest aircrafts
top_10_safest = safest_aircraft.head(10)


In [30]:
Aviation_Data.columns

NameError: name 'Aviation_Data' is not defined

In [28]:
plt.figure(figsize=(12,6))
plt.barh(top_10_safest['Make'] + ' ' + top_10_safest['Model'], top_10_safest['Fatal_Accident'], color='blue')
plt.xlabel('Number of Fatal Accidents')
plt.title('Top 10 Safest Aircraft Models (based on fatal accidents)')
plt.gca().invert_yaxis() 
plt.show()



NameError: name 'top_10_safest' is not defined

<Figure size 1200x600 with 0 Axes>

# 6. Business Recommendations

In [ ]:
# recommendation 1


Our analysis of aviation accidents from 1962–2023 shows that certain aircraft makes and models have consistently experienced zero fatal accidents and very few serious injuries despite having a significant number of flights.
 Prioritize purchasing aircraft such as [Example: Cessna 172], [Example: Piper PA-28], and [Example: Beechcraft Bonanza], which ranked in the Top 3 safest models according to historical data
 This reduces the company’s exposure to catastrophic risks and builds a reputation for safety in both commercial and private aviation operations.

# Suggest the three safest/lowest-risk aircraft based on the findings

    Small to mid-size general aviation aircraft (commonly used for training, leisure, and small charters) tend to have lower fatal accident rates compared to large commercial jets or experimental aircraft.

    Action:
    ✅ Start the aviation portfolio by operating small general aviation aircraft (4- to 8-seat planes) for charter services, private rentals, or flight schools.

    Benefit:
    Lower risk, easier regulatory compliance, and reduced initial investment costs, while building operational experience before scaling into larger commercial aircraft.

 # Provide recommendations for purchase strategy
 Prioritize Low-Risk Aircraft Models with No Fatal History

Insight:
From analysis of 1962–2023 accident data, several aircraft models (e.g., [Insert Top 3 Models]) show zero fatal accidents even across a significant number of incidents.

Action:
Begin fleet acquisition by targeting aircraft models that:

    Have zero historical fatal accidents.

    Have minimal serious injuries.

    Have more than 10 recorded incidents (ensuring sufficient reliability).

    Example aircraft models identified: [Example 1], [Example 2], [Example 3].
    


Recommendation 2: Focus on Aircraft Operating Primarily in Good Weather and Routine Phases

Insight:
Accidents are statistically lower risk when:

    Occurring during approach, landing, or taxiing (rather than cruise or takeoff).

    Under Visual Meteorological Conditions (VMC = good weather).

Action:
Select aircraft that are certified and proven for:

    Daylight VMC operations.

    Short-range, routine commercial flights where weather unpredictability is minimized.

This will naturally align the business with historically safer operations.

Recommendation 3: Implement Strict Maintenance and Pilot Training Programs on Acquired Models

Insight:
Most accidents (even non-fatal) are linked to:

    Mechanical failure due to poor maintenance.

    Pilot error especially during critical flight phases.

Action:
Develop standard operating procedures immediately for:

    Rigorous, scheduled aircraft maintenance tied to FAA best practices.

    Targeted pilot training programs emphasizing critical scenarios (e.g., emergency descent, engine failure).

Early investment in operations safety will reduce risk exposure and build client trust.

# 7. Prepare a Short, Business-Focused Storyline

A few slides  to guide the aviation division head.